### PR TITLE
Controllers, pages and sorting

### PR DESCRIPTION
**Pull request by NTanatar**

### CODE REVIEW

File: src/main/java/com/tannat/country/controllers/CityController.java [hunk 1]
```diff
@@ -1,4 +1,47 @@
...
+    public CityDto add(@RequestBody CityDto city) {
+        return cityService.add(city);
+    }
+
+    @PutMapping("/{id}")
+    public CityDto update(@PathVariable Long id,
+                          @RequestBody CityDto city) {
+        city.setId(id);
+        return cityService.update(city);
+    }
```
Comment (Andriireva) [inline src/main/java/com/tannat/country/controllers/CityController.java]: city is a sub resource of country. The url path should be /countries/<countryId>/cities/<cityId>

File: src/main/java/com/tannat/country/controllers/PageController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import com.tannat.country.services.CountryService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/v1/pages")
+@RequiredArgsConstructor
+public class PageController {
```
Comment (Andriireva) [inline src/main/java/com/tannat/country/controllers/PageController.java]: It is not required to add a separate controller for pagination. add optional request parameters variables to the get requests

**Review by Andriireva — CHANGES_REQUESTED** [review]

File: src/main/java/com/tannat/country/controllers/CityController.java (hunk 1 above)
Comment (NTanatar) [reply src/main/java/com/tannat/country/controllers/CityController.java]: changed

File: src/main/java/com/tannat/country/controllers/PageController.java (hunk 2 above)
Comment (NTanatar) [reply src/main/java/com/tannat/country/controllers/PageController.java]: ok

**Review by Andriireva — APPROVED** [review]

**Pull request merged**